tests/console_test.go

```diff
@@ -55,7 +52,9 @@ var _ = Describe("[rfe_id:127][posneg:negative][crit:medium][vendor:cnv-qe@redha
 
 	RunVMIAndWaitForStart := func(vmi *v1.VirtualMachineInstance) {
 		By("Creating a new VirtualMachineInstance")
-		Expect(virtClient.RestClient().Post().Resource("virtualmachineinstances").Namespace(util.NamespaceTestDefault).Body(vmi).Do(context.Background()).Error()).To(Succeed())
+		createdVmi, err := virtClient.VirtualMachineInstance(util.NamespaceTestDefault).Create(vmi)
```

Comment:
`tests.RunVMIAndExpectLaunch` is problematic because it creates a VMI with a hardcoded namespace (`NamespaceTestDefault`).

But except that, it will work for you if this is the namespace you need anyway.
The reason is: the creation function returns the new VMI object and then it is used to do all the other steps (waiting).

Reply:
how would it work? `tests.RunVMIAndExpectLaunch` does not have that `*vmi = *createdVmi` update. I would have to explicitly set the namespace inside `vmi.ObjectMeta` which you asked me to avoid.

I see only these options:
* keep `RunVMIAndWaitForStart`, for now.
* change `tests.RunVMIAndExpectLaunch` so it modifies its arg
* add a `WithDefaultNamespace()` option to `libvmi` and use it here repeatedly.

Reply:
[RunVMIAndWaitForStart](https://github.com/kubevirt/kubevirt/blob/811dd0a8ed81333c9bc8f535fab05f80c89c67f5/tests/utils.go#L1608) calls [RunVMI](https://github.com/kubevirt/kubevirt/blob/811dd0a8ed81333c9bc8f535fab05f80c89c67f5/tests/utils.go#L1595) which calls [Create](https://github.com/kubevirt/kubevirt/blob/811dd0a8ed81333c9bc8f535fab05f80c89c67f5/tests/utils.go#L1602) with the mentioned namespace, then returns an updated VMI to the caller. Then it calls the [Wait](https://github.com/kubevirt/kubevirt/blob/811dd0a8ed81333c9bc8f535fab05f80c89c67f5/tests/utils.go#L1611).

The only thing that is missing now, is that after the Wait*, the VMI is not up do date, so if someone wants to query something, it better first fetch it again using Get.

Reply:
> add a WithDefaultNamespace() option to libvmi and use it here repeatedly.

Setting the namespace in the spec explicitly is wrong, I do not know why you still want to do it.

Reply:
hmm, so sorry - I misspoke. `vmi = tests.RunVMIAndExpectLaunch(vmi, 30)` is fine.